docs/source/user/getting_started/advanced_workflow_concepts.rst

```diff
@@ -0,0 +1,309 @@
+    - path: inputs.conductance_value
+    values:
+    - 0
+    - 100
+    - 200
+    nesting_order: 0
+
+Groups
+-------
+
+To combine outputs from multiple elements, you can use a ``group`` in a task schema:
+
+.. code-block:: yaml
+
+  - objective: my_task_schema
+    inputs:
+    - parameter: p2
+        group: my_group
+
+combined with a ``groups`` entry in the task itself.
+
+.. code-block:: yaml
+
+  - schema: my_task_schema
+    groups:
+    - name: my_group
+
+
+Then whichever parameters are linked with the group in the task schema will be received by the task as a list.
+
+Here is an example workflow using sequences and groups that you might wish to run to solidify your understanding
+
+.. code-block:: yaml
+
+    # groups_workflow.yaml
+
+    template_components:
+      task_schemas:
+        - objective: s1
+          inputs:
+            - parameter: p1
+          outputs:
+            - parameter: p2
+          actions:
+            - commands:
+                - command: echo $(( <<parameter:p1>> + 1 )) # This is printed to stdout
+                - command: echo $(( <<parameter:p1>> + 1 )) # This is captured as p2
+                  stdout: <<int(parameter:p2)>>
+        - objective: s2
+          inputs:
+            - parameter: p2
+              group: my_group
+          outputs:
+            - parameter: p3
+          actions:
+            - commands:
+                - command: echo <<parameter:p2>> # This one is printed to stdout
+                - command: echo $(( <<sum(parameter:p2)>> )) # This is captured as p3
+                  stdout: <<int(parameter:p3)>>
+    tasks:
+      - schema: s1
+        sequences:
+          - path: inputs.p1
+            values: [1, 2]
+        groups:
+          - name: my_group
+      - schema: s2
+
+
+Task schema shortcuts
+---------------------
+
+Input file generators
+~~~~~~~~~~~~~~~~~~~~~
+
+``input_file_generators`` is a convenience shortcut for a python script which generates an input file
+for a subsequent action within a task. It's more compact, easier to reference, and has more interaction options.
+The first parameter in the input generator (python) function definition must be "path",
+which is the file path to ``input_file``, the file you want to create.
+Given this is a {{ app_name }} input file, the path is just the file name which will be created in the
+execute directory.
+The ``input_file`` must point to the label of a file in ``command_files``.
+``from_inputs`` defines which of the task schema inputs are required for each of the ``input_file_generators``.
+
+.. code-block:: yaml
+
+    task_schemas:
+    - objective: my_task_schema
+    actions:
+    - input_file_generators:
+      - input_file: my_command_file
+        from_inputs:
+        - my_input_1
+        - my_input_2
+        script: <<script:/full/path/to/generate_input_file.py>>
+
+An example is given in [advanced_workflow.yaml](advanced_workflow.yaml), along with the alternative code which would be needed
```

Comment:
This is a markdown URL.

Reply:
This was copied through MatFlow docs from [@gcapes 's docs](https://github.com/LightForm-group/matflow-user-documentation/blob/master/advanced_workflows.md) but [that yaml file](https://github.com/LightForm-group/matflow-user-documentation/blob/master/advanced_workflow.yaml) doesn't exist in this repo or in the MatFlow one. 

The file contains hardcoded paths on Gerard's machine so I'm not sure it should be made public on these docs, I'll remove it for now and make an issue to add something equivalent.